### PR TITLE
migrate default_preemption.go to structured logging

### PR DESCRIPTION
**Pull request by tanjing2020**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Ref:
https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1602-structured-logging
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```
NONE

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

### CODE REVIEW

File: pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go [hunk 1]
```diff
@@ -438,7 +440,9 @@ func SelectCandidate(candidates []Candidate) Candidate {
 	}
 
 	// We shouldn't reach here.
-	klog.Errorf("should not reach here, no candidate selected from %v.", candidates)
+	for i := range candidates {
+		klog.ErrorS(errors.New("Candidate is not selected"), "Should not reach here", "candidate", candidates[i])
+	}
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go:445]: can't we do: `klog.ErrorS(errors.New("no candidate selected"), "should not reach here", "candidates", candidates)`

File: pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go [hunk 2]
```diff
@@ -197,10 +199,10 @@ func (pl *DefaultPreemption) FindCandidates(ctx context.Context, state *framewor
 
 	potentialNodes := nodesWherePreemptionMightHelp(allNodes, m)
 	if len(potentialNodes) == 0 {
-		klog.V(3).Infof("Preemption will not help schedule pod %v/%v on any node.", pod.Namespace, pod.Name)
+		klog.V(3).InfoS("Preemption will not help schedule pod on any node", "pod", klog.KObj(pod))
 		// In this case, we should clean-up any existing nominated node name of the pod.
 		if err := util.ClearNominatedNodeName(pl.fh.ClientSet(), pod); err != nil {
-			klog.Errorf("Cannot clear 'NominatedNodeName' field of pod %v/%v: %v", pod.Namespace, pod.Name, err)
+			klog.ErrorS(err, "Cannot clear 'NominatedNodeName' field of pod", "pod", klog.KObj(pod))
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go:205]:
```suggestion
			klog.ErrorS(err, "cannot clear 'NominatedNodeName' field of pod", "pod", klog.KObj(pod))
```

File: pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go [hunk 3]
```diff
@@ -562,7 +566,7 @@ func pickOneNodeForPreemption(nodesToVictims map[string]*extenderv1.Victims) str
 		// Get earliest start time of all pods on the current node.
 		earliestStartTimeOnNode := util.GetEarliestPodStartTime(nodesToVictims[node])
 		if earliestStartTimeOnNode == nil {
-			klog.Errorf("earliestStartTime is nil for node %s. Should not reach here.", node)
+			klog.ErrorS(errors.New("EarliestStartTime is nil for node"), "Should not reach here", "node", node)
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go:569]:
```suggestion
			klog.ErrorS(errors.New("earliestStartTime is nil for node"), "should not reach here", "node", node)
```

File: pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go [hunk 4]
```diff
@@ -690,7 +694,7 @@ func selectVictimsOnNode(
 func PrepareCandidate(c Candidate, fh framework.Handle, cs kubernetes.Interface, pod *v1.Pod) error {
 	for _, victim := range c.Victims().Pods {
 		if err := util.DeletePod(cs, victim); err != nil {
-			klog.Errorf("Error preempting pod %v/%v: %v", victim.Namespace, victim.Name, err)
+			klog.ErrorS(err, "Error preempting pod", "pod", klog.KObj(victim))
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go:697]:
```suggestion
			klog.ErrorS(err, "preempting pod", "pod", klog.KObj(victim))
```

File: pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go [hunk 5]
```diff
@@ -708,7 +712,7 @@ func PrepareCandidate(c Candidate, fh framework.Handle, cs kubernetes.Interface,
 	// lets scheduler find another place for them.
 	nominatedPods := getLowerPriorityNominatedPods(fh.PreemptHandle(), pod, c.Name())
 	if err := util.ClearNominatedNodeName(cs, nominatedPods...); err != nil {
-		klog.Errorf("Cannot clear 'NominatedNodeName' field: %v", err)
+		klog.ErrorS(err, "Cannot clear 'NominatedNodeName' field")
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go:715]:
```suggestion
		klog.ErrorS(err, "cannot clear 'NominatedNodeName' field")
```

File: pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go [hunk 6]
```diff
@@ -125,15 +126,16 @@ func (pl *DefaultPreemption) preempt(ctx context.Context, state *framework.Cycle
 	// It's safe to directly fetch pod here. Because the informer cache has already been
 	// initialized when creating the Scheduler obj, i.e., factory.go#MakeDefaultErrorFunc().
 	// However, tests may need to manually initialize the shared pod informer.
+	podNamespace, podName := pod.Namespace, pod.Name
 	pod, err := pl.podLister.Pods(pod.Namespace).Get(pod.Name)
 	if err != nil {
-		klog.Errorf("Error getting the updated preemptor pod object: %v", err)
+		klog.ErrorS(err, "Error getting the updated preemptor pod object", "pod", klog.KRef(podNamespace, podName))
```
[the remark this answers is not in the document]
Comment (tanjing2020) [reply pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go]: when err != nil, pod is nil. In this case, we cannot get the pod information.

File: pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go (hunk 1 above)
Comment (tanjing2020) [reply pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go]: Thanks for your advice, I'll fix it.

**Comment by tanjing2020** [comment]
Thanks for your advices, I fixed everything except the first one. Please check it again, thanks. 
/recc @ahg-g

File: pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go (hunk 1 above)
Comment (ahg-g) [reply pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go]: can you do this one?

Comment (tanjing2020) [reply pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go]: I didn't understand it correctly yesterday. I have revised it.

**Pull request merged**